bot/cogs/tags.py

```diff
@@ -84,6 +84,66 @@ def __init__(self, bot: AutoShardedBot):
 
         return tag_data
 
+    @staticmethod
+    async def validate(embed: Embed, author: User, tag_name: str, tag_content: str = None) -> bool:
+        """
+        Edit an embed based on the validity of a tag name
+
+        :param embed: The embed to modify.
+        :param author: The user that called the command
+        :param tag_name: The name of the tag to validate.
+        :param tag_content: The tag's content, if any.
+        :return: True if tag details passed all checks, otherwise False.
+        """
+
+        def is_number(string):
+            try:
+                float(string)
+            except ValueError:
+                return False
+            else:
+                return True
+
+        # Replace any special characters
+        raw_name = tag_name.translate(
+            {
+                0x8: "\\b",  # Backspace
+                0x9: "\\t",  # Horizontal tab
+                0xA: "\\n",  # Linefeed
+                0xB: "\\v",  # Vertical tab
+                0xC: "\\f",  # Form feed
+                0xD: "\\r"   # Carriage return
+            }
+        )
+
+        # 'tag_name' has any special characters
+        if tag_name != raw_name:
+            log.warning(f"{author} tried to put a special character in a tag name. "
+                        "Rejecting the request.")
+            embed.title = "Please don't do that"
+            embed.description = "Don't be ridiculous, special characters obviously aren't allowed in the tag name"
+
+        # 'tag_content' (when not None) or 'tag_name' consists of nothing but whitespace
+        elif (tag_content and not tag_content.strip()) or not tag_name.strip():
+            log.warning(f"{author} tried to create a tag with a name consisting only of whitespace. "
+                        "Rejecting the request.")
+            embed.title = "Please don't do that"
+            embed.description = "Tags should not be empty, or filled with whitespace."
+
+        # 'tag_name' is a number of some kind, we don't allow that.
+        elif is_number(tag_name):
+            log.error("inside the is_number")
+            log.warning(f"{author} tried to create a tag with a digit as its name. "
+                        "Rejecting the request.")
+            embed.title = "Please don't do that"
+            embed.description = "Tag names can't be numbers."
+
+        # No checks failed, return True (valid)
+        else:
+            return True
```

Comment:
I don't like this because passing in stuff "by reference" and modifying it in your helper function is kinda implicit and C-ey. I'd prefer if you simply created and returned an embed instead.